### DISCUSSION TITLE
does it resized the sliced images when SAHI ?

### DISCUSSION THREAD

**Discussion by Jasper-daoheng** [obss/sahi#885] [Q&A] [answered]
The paper describes that the image will be sliced into small patches and then resized into a bigger picture when during SAHI, but in the code, I do not find the patches that are resized. For example, I use yolov5 to infer and debug, the hyperparameter is default, width and height is 256 , I print the shape of the picture that feeds into the model, and it shows (256,256,3)..... So, I wonder whether the paper is different from the code implementation or if I missing something else.

**Comment by fcakyon** [comment, marked as answer]
No implementation is the same, resizing is happening inside yolo model, you can set the resize size with image_size parameter in sahi https://github.com/obss/sahi/blob/0077a143abdabad2ac81eb375377c744662f7578/sahi/models/yolov5.py#L65

**Reply by Jasper-daoheng** [reply]
It seems that I understand your meaning, the yolov5 model will resize the input image automatically. Although the image shape is (256,256,3) after slicing and feeding into the yolov5 model, the yolov5 will resize the (256,256,3) into a bigger image, am I right?

**Reply by fcakyon** [reply]
Yes you are right, it will take 256 slices, interpolate it to image_size (640 by default), perform inference in 640, then map it back to 256 and merge all slice predictions.

**Reply by Jasper-daoheng** [reply]
thanks a lot for your general reply, but I still have one question. The paper indicates that "patches are resized by preserving the aspect ratio so that image width is between 800 to 1333 pixels", so if I slice an image with a default hyperparameter, which width and height are 256. since yolov5 's default input image size is 640, does this mean I need to revise the default input image size in Yolo? If true, how can I revise the default input image size in yolov5?
